app/src/main/java/org/connectbot/HostListActivity.java

```diff
@@ -394,13 +431,37 @@ public void onClick(DialogInterface dialog, int which) {
 							hostdb.deleteHost(host);
 							updateHandler.sendEmptyMessage(-1);
 						}
-						})
+					})
 					.setNegativeButton(R.string.delete_neg, null).create().show();
 
 				return true;
 			}
 		});
 	}
+	/**
+	 * Disconnects all active connections and closes the activity if appropriate.
+	 */
+	private void disconnectAll() {
+		// TODO(jklein24): Show a confirm dialog before actually disconnecting.
+		if (bound == null) {
+			waitingForDisconnectAll = true;
+			return;
+		}
+		// Copy the bridges list because bridges are removed from the array when disconnected.
+		for (TerminalBridge bridge : new ArrayList<TerminalBridge>(bound.bridges)) {
+			bridge.dispatchDisconnect(true);
+		}
+		updateHandler.sendEmptyMessage(-1);
+		waitingForDisconnectAll = false;
+
+		if (closeOnDisconnectAll) {
+			// Clear the intent so that the activity can be relaunched without closing.
+			// TODO(jlklein): Find a better way to do this.
```

Comment:
Perhaps, but then should that also be the case if the user disconnects individually from all connections using the context menus? I feel like if the user was already in the hostlistactivity and then presses disconnect all, they should stay where they were. I could be convinced either way, though. 

sigh... totally missed that there's already a `disconnectAll` in TerminalManager. Updated.